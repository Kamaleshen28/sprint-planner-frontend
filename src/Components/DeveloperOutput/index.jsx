import * as React from 'react';
import PropTypes from 'prop-types';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import Story from '../StoryOutput';
import { IconButton, Typography } from '@mui/material';
import Modal from '@mui/material/Modal';
import Tile from '../Tile';
import CloseIcon from '@mui/icons-material/Close';
import { createTheme, responsiveFontSizes } from '@mui/material/styles';

const Developer = (props) => {
  const { content, developerName } = props;
  const [open, setOpen] = React.useState(false);
  const handleOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  const style = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 400,
    bgcolor: '#eeeeee',
    border: '2px solid lightgrey',
    pt: 2,
    px: 4,
    pb: 3,
    overflowY: 'auto',
    minHeight: '100px',
  };
  let theme = createTheme({
    typography: {
      fontSize: 10,
      fontWeightBold: 400,
    },
  });
  theme = responsiveFontSizes(theme);
  return (
    <React.Fragment>
      <Tile handleOpen={handleOpen}> {developerName}</Tile>
      <Modal
        open={open}
        onClose={handleClose}
        aria-labelledby="child-modal-title"
        aria-describedby="child-modal-description"
      >
        <Box
          width={2 / 3}
          p={3}
          sx={{ ...style, overflow: 'auto', borderRadius: '4px' }}
          maxHeight={300}
        >
          <Box
            display="flex"
            sx={{
              position: 'relative',
              bgcolor: 'black',
              color: 'white',
              width: '100%',
              borderRadius: '6px',
              borderBottomLeftRadius: 0,
              borderBottomRightRadius: 0,
              mb: 2,
              // borderTopRightRadius: 5,
              height: 50,
            }}
          >
            <Typography variant="h4" theme={theme} m="auto">
              {developerName}
            </Typography>
            <IconButton
              aria-label="settings"
              onClick={handleClose}
              sx={{
                position: 'absolute',
                right: 0,
                '&:hover': { bgcolor: '#f9f9f936' },
                p: 0,
                height: '40px',
                mt: '5px',
                mr: '5px',
              }}
            >
              <CloseIcon
                sx={{
                  color: 'white',
                  p: 1,
                }}
              />
            </IconButton>
          </Box>
          <Grid container spacing={2}>
            {content.map((story, index) => (
              <Grid item xs={12} key={index}>
                <Story
                  title={story.title}
                  id={story.id + 1}
                  dependencies={story.dependencies}
                  startDay={story.startDay}
                  endDay={story.endDay}
                  developers={story.developers}
                  assignedDeveloperId={story.assignedDeveloperId}
                />
              </Grid>
            ))}
          </Grid>
        </Box>
      </Modal>
    </React.Fragment>
  );
};
Developer.propTypes = {
  developerName: PropTypes.string,
  content: PropTypes.array,
};
export default Developer;
