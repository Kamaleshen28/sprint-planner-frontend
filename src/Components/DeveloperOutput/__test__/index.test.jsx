import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import DeveloperOutput from '..';

describe('DeveloperOutput', () => {
  it('should render correctly', () => {
    const mockProps = {
      developerName: '',
      content: [],
    };
    const { asFragment } = render(<DeveloperOutput {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
