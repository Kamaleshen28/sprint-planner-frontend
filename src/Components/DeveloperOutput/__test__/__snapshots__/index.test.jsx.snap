// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeveloperOutput should render correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-x0rgt9-MuiButtonBase-root-MuiButton-root"
    data-testid="modal-button"
    tabindex="0"
    type="button"
  >
     
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
